internal/pkg/manifest/initial_manifest_integration_test.go

```diff
@@ -0,0 +1,413 @@
+		},
+		"with retries and no timeout": {
+			inProps: ScheduledJobProps{
+				WorkloadProps: &WorkloadProps{
+					Name:       "cuteness-aggregator",
+					Dockerfile: "./cuteness-aggregator/Dockerfile",
+				},
+				Platform: PlatformArgsOrString{
+					PlatformString: nil,
+					PlatformArgs:   PlatformArgs{},
+				},
+				Schedule: "@every 5h",
+				Retries:  5,
+			},
+			wantedTestData: "scheduled-job-no-timeout.yml",
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			// GIVEN
+			path := filepath.Join("testdata", tc.wantedTestData)
+			wantedBytes, err := ioutil.ReadFile(path)
+			require.NoError(t, err)
+			manifest := NewScheduledJob(&tc.inProps)
+
+			// WHEN
+			tpl, err := manifest.MarshalBinary()
+			require.NoError(t, err)
+
+			// THEN
+			require.Equal(t, string(wantedBytes), string(tpl))
+		})
+	}
+}
+
+func TestEnvironment_InitialManifestIntegration(t *testing.T) {
+	testCases := map[string]struct {
+		inProps        EnvironmentProps
+		wantedTestData string
+	}{
+		"fully configured with customized vpc resources": {
+			inProps: EnvironmentProps{
+				Name: "test",
+				CustomConfig: &config.CustomizeEnv{
+					VPCConfig: &config.AdjustVPC{
+						CIDR:               "mock-cidr-0",
+						AZs:                []string{"mock-az-1", "mock-az-2"},
+						PublicSubnetCIDRs:  []string{"mock-cidr-1", "mock-cidr-2"},
+						PrivateSubnetCIDRs: []string{"mock-cidr-3", "mock-cidr-4"},
+					},
+					ImportCertARNs:     []string{"mock-cert-1", "mock-cert-2"},
+					InternalALBSubnets: []string{"mock-subnet-id-3", "mock-subnet-id-4"},
+				},
+				Telemetry: &config.Telemetry{
+					EnableContainerInsights: false,
+				},
+			},
+			wantedTestData: "environment-adjust-vpc.yml",
+		},
+		"fully configured with imported vpc resources": {
+			inProps: EnvironmentProps{
+				Name: "test",
+				CustomConfig: &config.CustomizeEnv{
+					ImportVPC: &config.ImportVPC{
+						ID:               "mock-vpc-id",
+						PublicSubnetIDs:  []string{"mock-subnet-id-1", "mock-subnet-id-2"},
+						PrivateSubnetIDs: []string{"mock-subnet-id-3", "mock-subnet-id-4"},
+					},
+					ImportCertARNs: []string{"mock-cert-1", "mock-cert-2"},
+				},
+				Telemetry: &config.Telemetry{
+					EnableContainerInsights: true,
+				},
+			},
+			wantedTestData: "environment-import-vpc.yml",
+		},
+		"basic manifest": {
+			inProps: EnvironmentProps{
+				Name: "test",
+			},
+			wantedTestData: "environment-default.yml",
+		},
+	}
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			// GIVEN
+			path := filepath.Join("testdata", tc.wantedTestData)
+			wantedBytes, err := ioutil.ReadFile(path)
+			require.NoError(t, err)
+			manifest := NewEnvironment(&tc.inProps)
+
+			// WHEN
+			tpl, err := manifest.MarshalBinary()
+			require.NoError(t, err)
+
+			// THEN
+			require.Equal(t, string(wantedBytes), string(tpl))
+		})
+	}
+}
+
+func TestPipelineManifest_InitialManifest_Integration(t *testing.T) {
```

Comment:
Although GH has hidden all the diffs in this file, this test suite is actually relevant.